### PR TITLE
Implementing "jsonMode" for LLM calls, and from the react app iframe calls

### PR DESCRIPTION
**Pull request by jakedahn**
This PR introduces `jsonMode` capability to the toolshed llm api endpoints.

Unfortunately, each language model provider has a different strategy for allowing the return of json data. We primarily use Claude and Groq at the moment, claude only supports prompting (and they provide some instructions in their [docs for how to do this better](https://docs.anthropic.com/en/docs/test-and-evaluate/strengthen-guardrails/increase-consistency)), and in groq you must prompt but can also require that the response is valid json.

So this PR introduces some special casing prompt additions to various models when `jsonMode: true` is set as an api request parameter.

I also removed a handful of never-used language models to clean up some dependencies and surfaces area we need to support.

### CODE REVIEW

**Comment by jakedahn** [comment]
This should fix https://github.com/commontoolsinc/labs/pull/965

File: charm/src/iframe/static.ts [hunk 1]
```diff
@@ -673,12 +647,23 @@ async function fetchLLMResponse() {
...
 }
 \`\`\`
 
+If you need JSON to be returned from the LLM, you can enable the \`jsonMode\` flag in the \`promptPayload\`.
+
+\`\`\`jsx
+const promptPayload = {
+  system: 'Translate all the messages to emojis, reply in JSON.',
+  messages: ['Hi', 'How can I help you today?', 'tell me a joke'],
+  jsonMode: true
```
Comment (bfollington) [inline charm/src/iframe/static.ts:661]: consider: `mode: 'json'`? seems more common to me

File: toolshed/routes/ai/llm/generateText.ts [hunk 2]
```diff
@@ -12,13 +22,105 @@ export interface GenerateTextParams {
...
   stream?: ReadableStream;
 }
 
+// Configure the model parameters for JSON mode based on provider
+function configureJsonMode(
+  streamParams: Record<string, any>,
+  modelName: string,
+  messages: { role: "user" | "assistant"; content: string }[],
+  isStreaming: boolean,
+): void {
```
Comment (bfollington) [inline toolshed/routes/ai/llm/generateText.ts:39]: Thoughts on adding tests for this? It's got a bunch of specific edge cases so tests would act as documentation of expected behavior etc.

Might be generateable?

**Review by bfollington — CHANGES_REQUESTED** [review]
Not a big deal, but a couple comments.

You can merge without my re-review.

File: toolshed/routes/ai/llm/generateText.test.ts [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+// Import the functions for testing
+import { cleanJsonResponse, configureJsonMode } from "./generateText.ts";
+
+if (env.ENV !== "test") {
+  throw new Error("ENV must be 'test'");
+}
+
+Deno.test("JSON mode configuration", async (t) => {
+  // Test for Groq models
+  await t.step("Configures JSON mode correctly for Groq models", () => {
```
Comment (jsantell) [inline toolshed/routes/ai/llm/generateText.test.ts:13]: The `Deno.test`/`t.step` interface was used in basic-flow to handle some integration/astral slowness/assertions during development, but could probably be ported back to the BDD style -- if we don't have an explicit reason to use the inner `t.step` interface, consider the BDD wrapper around stepped tests e.g. `import { describe, it, beforeEach } from "@std/testing/bdd"`

File: toolshed/deno.json [hunk 4]
```diff
@@ -9,9 +9,7 @@
   },
   "imports": {
     "@/": "./",
-    "@ai-sdk/amazon-bedrock": "npm:@ai-sdk/amazon-bedrock@^1.1.6",
     "@ai-sdk/anthropic": "npm:@ai-sdk/anthropic@^1.1.6",
-    "@ai-sdk/cerebras": "npm:@ai-sdk/cerebras@^0.1.8",
```
Comment (jsantell) [inline toolshed/deno.json:14]: :+1: slimmer deps!

File: toolshed/routes/ai/llm/llm.handlers.ts [hunk 5]
```diff
@@ -56,21 +92,34 @@ export const getModels: AppRouteHandler<GetModelsRoute> = (c) => {
...
+
   try {
-    // Check cache for existing response
-    const cacheKey = await cache.hashKey(JSON.stringify(payload));
-    const cachedResult = await cache.loadItem(cacheKey);
-    if (cachedResult) {
-      const lastMessage =
-        cachedResult.messages[cachedResult.messages.length - 1];
-      return c.json(lastMessage);
+    // Check cache for existing response - skip caching for JSON mode requests
```
Comment (anotherjesse) [inline toolshed/routes/ai/llm/llm.handlers.ts:114]: is there a reason we want to skip caching for json requests?

File: toolshed/routes/ai/llm/llm.handlers.ts [hunk 6]
```diff
@@ -80,10 +129,14 @@ export const generateText: AppRouteHandler<GenerateTextRoute> = async (c) => {
     });
 
     if (!payload.stream) {
-      await cache.saveItem(cacheKey, {
-        ...payload,
-        messages: [...payload.messages, result.message],
-      });
+      // Save to cache only if not in JSON mode
+      if (!payload.mode) {
+        const cacheKey = await cache.hashKey(JSON.stringify(payload));
```
Comment (anotherjesse) [inline toolshed/routes/ai/llm/llm.handlers.ts:134]: ahh, this should be the cacheKey earlier - don't recompute it!
